kiota_abstractions/store/backed_model.py

```diff
@@ -16,6 +16,9 @@ class BackedModel:
     # Stores model information.
     backing_store: BackingStore
 
+    def __post_init__(self):
```

Comment:
this is usually done by the poxy.
Also, I don't know much about python, but this should be an interface (unless python allows for multi parents inheritance)

Reply:
> Also, I don't know much about python, but this should be an interface (unless python allows for multi parents inheritance)
It does e.g `class Entity(AdditionalDataHolder, BackedModel, Parsable):`

> this is usually done by the poxy.
Yes, after serialization. The issue here is during instantiation. Instead of `is_initialization_completed` being `True` by default in the backing store, we want it set to T`rue` only after initialization is completed using the `__post_init__` method. That way we can track changes to the fields thereafter.

Reply:
The same applies to Python: https://github.com/microsoft/kiota-abstractions-python/blob/554697a71fdaa92e9479808877dc244c0b89ed60/kiota_abstractions/store/backing_store_parse_node_factory.py#L23

The bug is unique to Python(because of how it instantiates dataclasses.) and occurs at https://github.com/microsoft/kiota-abstractions-python/blob/554697a71fdaa92e9479808877dc244c0b89ed60/kiota_abstractions/store/in_memory_backing_store.py#L23 

Say you have a class `Event` which is a backed model

```py
@dataclass
class Event(BackedModel):
    additional_data: Dict[str, Any] = field(default_factory=dict)
    id: Optional[str] = None
    odata_type: Optional[str] = "#microsoft.graph.event"
    attachments: Optional[List[Attachment]] = None
    attendees: Optional[List[Attendee]] = None
    body: Optional[ItemBody] = None
    calendar: Optional[Calendar] = None
    end: Optional[DateTimeTimeZone] = None
    has_attachments: Optional[bool] = None
    subject: Optional[str] = None
```
Currently, during instantiation, Python will first set the backing store property in the base class `BackedModel` and then proceed to set the default values in the `Event` class in order. What this means is that all these properties will be marked as changed since the backing store was already set and `is_initialization_completed` set to True by default.
Thus:

```cmd
event = Event()
event.backing_store.return_only_changed_values = True
event.backing_store.enumerate_()
```
Returns: `[('additional_data', {}),('id', None), ('odata_type', "#microsoft.graph.event"),('attachments', None), ('attendees', None), ('body', None), ('calendar', None), ('end', None), ('has_attachments', None), ('subject', None)]`

This is not correct since no value has changed after instantiation. 

What am proposing is, in the backing we set the default value of `is_initialization_completed` to False. The we wait until initialization has completed and the default values have been set then we use the `__post_init__` method to set it to True. That way 

```cmd
event = Event()
event.backing_store.return_only_changed_values = True
event.backing_store.enumerate_()
```
Returns: `[]` since no value has changed

And 

``cmd
event = Event()
event.subject = "Hello World"
event.backing_store.return_only_changed_values = True
event.backing_store.enumerate_()
```
Returns: `[('subject', "Hello World")]` since subject has changed after initialization.

Reply:
alright, thanks for the additional context, I'm guessing this is due to the way Python instantiates objects that varies from other languages.